validator/project_validator.go

```diff
@@ -1528,29 +1554,40 @@ func bvsWithTasksThatCallCommand(p *model.Project, cmd string) (map[string]map[s
 			}
 		}
 	}
-	return bvToTasksWithCmds, catcher.Resolve()
+	return bvToTasksWithCmds, numCmds, catcher.Resolve()
 }
 
 // validateTaskSyncCommands validates project's task sync commands.  In
 // particular, s3.push should be called at most once per task and s3.pull should
 // refer to a valid task running s3.push.  It does not check that the project
-// settings allow task syncing - see validateTaskSyncSettings.
-func validateTaskSyncCommands(p *model.Project) ValidationErrors {
+// settings allow task syncing - see validateTaskSyncSettings. If run long isn't set,
+// we don't validate dependencies if there are too many commands.
+func validateTaskSyncCommands(p *model.Project, runLong bool) ValidationErrors {
 	errs := ValidationErrors{}
 
 	// A task should not call s3.push multiple times.
 	s3PushCalls := p.TasksThatCallCommand(evergreen.S3PushCommandName)
 	errs = append(errs, validateTimesCalledPerTask(p, s3PushCalls, evergreen.S3PushCommandName, 1, Warning)...)
 
-	bvToTaskCmds, err := bvsWithTasksThatCallCommand(p, evergreen.S3PullCommandName)
+	bvToTaskCmds, numCmds, err := bvsWithTasksThatCallCommand(p, evergreen.S3PullCommandName)
 	if err != nil {
 		errs = append(errs, ValidationError{
 			Level:   Error,
 			Message: fmt.Sprintf("could not generate map of build variants with tasks that call command '%s': %s", evergreen.S3PullCommandName, err.Error()),
 		})
 	}
 
-	tvToTaskUnit := tvToTaskUnit(p)
+	checkDependencies := numCmds <= maxTaskSyncCommandsForDependenciesCheck || runLong
```

Comment:
I think the CLI flag won't be respected if the project yaml has < 300 tasks and they explicitly set `--long=false`. Maybe it's worth mentioning in the CLI usage text that slow checks will run no matter what `--long` is set to if the check can be done quickly?

Reply:
Makes sense although weird to word; attempted a fix